File: cpp/sorting/Bogo_Sort.cpp
#include<bits/stdc++.h>
using namespace std;

void display(int * array, int size) {
  for (int value = 0; value < size; value++)
    cout << array[value] << " ";
  cout << endl;
}

// To check if array is sorted or not 
bool isSorted(int array[], int n) {
  while (--n > 1)
    if (array[n] < array[n - 1])
      return false;
  return true;
}

// To generate permuatation of the array 
void shuffle(int array[], int n) {
  for (int i = 0; i < n; i++)
    swap(array[i], array[rand() % n]);
}

void bogoSort(int array[], int n) {
  while (!isSorted(array, n))
    shuffle(array, n);
}

// Driver code 
int main() {
  int n;
  cout << "Enter the number of elements: ";
  cin >> n;
  int arr[n];
  cout << "Enter the elements value:" << endl;
  for (int i = 0; i < n; i++) {
    cin >> arr[i];
  }
  cout << "Array before Bogo Sorting: ";
  display(arr, n);
  bogoSort(arr, n);
  cout << "Array after Bogo Sorting: ";
  display(arr, n);
}

File: python/sorting/bogo_sort.py
#Bogo Sort or Permutation Sort
# algorithm based on generate and test paradigm
#successively generates permutations of its input until it finds one that is sorted.
# Python program for implementation of Bogo Sort

import random

# Sorts array a[0..n-1] using Bogo sort
def bogoSort(a):
    n = len(a)
    while (is_sorted(a)== False):
        shuffle(a)

# To check if array is sorted or not
def is_sorted(a):
    n = len(a)
    for i in range(0, n-1):
        if (a[i] > a[i+1] ):
            return False
    return True

# To generate permuatation of the array
def shuffle(a):
    n = len(a)
    for i in range (0,n):
        r = random.randint(0,n-1)
        a[i], a[r] = a[r], a[i]

# Driver code to test above
a = []
n = int(input("Enter no of elements:"))
for i in range(0,n):
    element = int(input())
    a.append(element)

    
bogoSort(a)
print("Sorted array :")
for i in range(len(a)):
    print ("%d" %a[i]),
